nntrainer/src/optimizer.cpp

```diff
@@ -34,73 +34,25 @@
 
 namespace nntrainer {
 
-Optimizer::Optimizer(const OptType t, const OptParam p) {
-  type = t;
-  popt = p;
+int Optimizer::initialize(std::shared_ptr<Weight> weight_list,
+                          unsigned int num_weights, bool set_tensor) {
+  return ML_ERROR_NONE;
 }
 
-int Optimizer::setType(OptType t) {
-  int status = ML_ERROR_NONE;
-  if (t == OptType::unknown) {
-    ml_loge("Error: Optimizer is unknown");
-    return ML_ERROR_INVALID_PARAMETER;
-  }
-  type = t;
-  return status;
-}
+double Optimizer::getLearningRate(int iteration) {
+  double ll = learning_rate;
 
-int Optimizer::setOptParam(OptParam p) {
-  int status = ML_ERROR_NONE;
-  if (p.learning_rate <= 0) {
-    ml_loge("Error: learning_rate should be grater than 0 (%f)",
-            p.learning_rate);
-    return ML_ERROR_INVALID_PARAMETER;
+  if (decay_steps != -1) {
+    ll = ll * pow(decay_rate, (iteration / decay_steps));
```

Comment:
`powf` for the precision?

Reply:
pow operates on double, powf operates on float.

Reply:
No, https://github.com/nnstreamer/nntrainer/pull/633#discussion_r503057084 this explained Thanks.